conans/cli/api/subapi/install.py

```diff
@@ -0,0 +1,146 @@
+        @param install_folder:
+        @param base_folder: Tipically current folder
+        @param conanfile_folder: Folder where the conanfile is located
+        @param build_modes:
+        @param generators:
+        @param no_imports:
+        @param create_reference:
+        @param update:
+        @param test:
+        """
+        app = ConanApp(self.conan_api.cache_folder)
+        # FIXME: remote_name should be remote
+        app.load_remotes([Remote(remote_name, None)], update=update)
+
+        out = ConanOutput()
+
+        root_node = deps_graph.root
+        conanfile = root_node.conanfile
+
+        # TODO: maybe we want a way to get this directly from the DepsGraph?
+        reference = deps_graph.nodes[1] if root_node.recipe == RECIPE_VIRTUAL else None
+
+        if root_node.recipe == RECIPE_VIRTUAL:
+            out.highlight("Installing package: %s" % str(reference))
+        else:
+            conanfile.output.highlight("Installing package")
+        print_graph(deps_graph)
+
+        installer = BinaryInstaller(app)
+        # TODO: Extract this from the GraphManager, reuse same object, check args earlier
+        build_modes = BuildMode(build_modes)
+        installer.install(deps_graph, build_modes)
+
+        root_node = deps_graph.root
+        conanfile = root_node.conanfile
+
+        if hasattr(conanfile, "layout") and not test:
+            conanfile.folders.set_base_install(conanfile_folder)
+            conanfile.folders.set_base_imports(conanfile_folder)
+            conanfile.folders.set_base_generators(conanfile_folder)
+        else:
+            conanfile.folders.set_base_install(install_folder)
+            conanfile.folders.set_base_imports(install_folder)
+            conanfile.folders.set_base_generators(base_folder)
+
+        if install_folder:
+            # Write generators
+            tmp = list(conanfile.generators)  # Add the command line specified generators
+            generators = set(generators) if generators else set()
+            tmp.extend([g for g in generators if g not in tmp])
+            conanfile.generators = tmp
+            write_generators(conanfile)
+
+            if not no_imports:
+                run_imports(conanfile)
+            if type(conanfile).system_requirements != ConanFile.system_requirements:
+                call_system_requirements(conanfile)
+
+            if not create_reference and reference:
+                # The conanfile loaded is a virtual one. The one w deploy is the first level one
+                neighbours = deps_graph.root.neighbors()
+                deploy_conanfile = neighbours[0].conanfile
+                if hasattr(deploy_conanfile, "deploy") and callable(deploy_conanfile.deploy):
+                    run_deploy(deploy_conanfile, install_folder)
+
+        return deps_graph
+
+    # maybe this is not wanted as API method, for the moment just leaving this to not
+    # explode things too much
```

Comment:
Yes, certainly we don't want this as API method, but it is fine as-is at the moment if it helps to move forward.